### PR TITLE
BUG: CCITTFaxDecode crashed

### PR DESCRIPTION
**Pull request by hoshantm**
The following statements failed:
- decodeParms.get("/K", 1)
  get method not found.

- width = decodeParms["/Columns"]
  decodeParms[0]["/Columns"] works.

If the original code works for some file, then the type of decodeParms could be checked and accordingly the original code or new code could be invoked.

### CODE REVIEW

File: PyPDF2/filters.py [hunk 1]
```diff
@@ -167,12 +167,12 @@ def encode(data):
         return compress(data)
     encode = staticmethod(encode)
 
-
 class ASCIIHexDecode(object):
     def decode(data, decodeParms=None):
         retval = ""
         char = ""
         x = 0
+        data = data.decode()
```
Comment (MartinThoma) [inline PyPDF2/filters.py:175]: What issue does this solve?

**Comment by MartinThoma** [comment]
Thank you for your contribution! The issue that decodeParms can (not must!) be an array was already resolved. Is it ok for you if I fix the merge conflicts in `filters.py`?

**Comment by hoshantm** [comment]
> Thank you for your contribution! The issue that decodeParms can (not must!) be an array was already resolved. Is it ok for you if I fix the merge conflicts in `filters.py`?

Not being familiar with the code, my solution was probably quick and dirty. So, please go ahead and implement the necessary change in filters.py.

**Comment by MartinThoma** [comment]
I need help on the remaining `data = data.decode()` line. Why is that necessary?

**Comment by hoshantm** [comment]
It has been a long time with just a couple of files causing a crash during
decoding. I kind of handled it quick and dirty to get things going. I will
try to reproduce the problem but I need some time.

**Comment by MartinThoma** [comment]
As I cannot reproduce any issue this fixes, I'll close it. If anybody runs into the same problem, I can review it again :-)

**Pull request closed without merging**